### PR TITLE
RN-97 Changed navigation from NavigationExperimental to ReactNative Navigation

### PR DESCRIPTION
**Pull request by enahum**
#### Summary
This PR changes the whole navigation that we were using with NavigationExperimental (deprecated) and moved to [React Native Navigation](https://wix.github.io/react-native-navigation/)

Important notes:
* `scenes` folder was moved to `screens` folder
* Navigation now uses the native Platform specific navigation, that means that it behaves differently in IOS and Android in terms of transitions.
* Sometimes we need to use a custom header so this PR tries to be as closed as the native component for each Platform.

**Know issues**:
* When tapping on a push notification and the app is in cold state, the app opens but does not navigate to the channel were the notification is from.

#### Ticket Link
https://mattermost.atlassian.net/browse/RN-97

#### Checklist
[Place an '[x]' (no spaces) in all applicable fields. Please remove unrelated fields.]
- [x] Added or updated unit tests (required for all new features)
- [x] Has UI changes

#### Device Information
This PR was tested on:
* IOS 10.3.1 iPhone 6s
* Android 6.0.1 Samsung J5

### CODE REVIEW

File: app/screens/edit_post/edit_post.js [hunk 1]
```diff
@@ -166,12 +140,55 @@ export default class EditPost extends PureComponent {
...
+            </TouchableOpacity>
+        );
+
+        const navbarTitle = (
+            <FormattedText
+                id='mobile.edit_post.title'
+                defaultMessage='Editing Message'
+                ellipsizeMode='tail'
+                numberOfLines={1}
+                style={{
```
Comment (jarredwitt) [inline app/screens/edit_post/edit_post.js]: Does this need the android styling like the others?

File: app/screens/account_notifications/account_notifications.js [hunk 2]
```diff
@@ -493,14 +516,52 @@ export default class AccountNotifications extends PureComponent {
...
+                labelDefaultMessage='Save'
+                labelId='edit_post.save'
+                loadingEventName='saving_notify_props'
+            />
+        );
+
         return (
             <View style={style.wrapper}>
+                <StatusBar barStyle='light-content'/>
+                <NavBar
```
Comment (enahum) [inline app/screens/account_notifications/account_notifications.js]: The loading indicator should show in these, maybe we need to fix the styling for both IOS and Android, don't you agree? that unless we get rid of the load indicators on the right.

I'm ok with both options

File: app/screens/account_notifications/account_notifications.js [hunk 3]
```diff
@@ -536,9 +601,16 @@ const getStyleSheet = makeStyleSheetFromTheme((theme) => {
         scrollViewContent: {
             paddingBottom: 30
         },
-        wrapper: {
-            flex: 1,
-            backgroundColor: theme.centerChannelBg
+        navTitle: {
```
Comment (enahum) [inline app/screens/account_notifications/account_notifications.js]: yes, probably best if we end up keeping that NavBar component

File: app/screens/edit_post/edit_post.js (hunk 1 above)
Comment (enahum) [inline app/screens/edit_post/edit_post.js]: I did add it before but it looked really crapy hehe

**Comment by enahum** [comment]
@jarredwitt changes made and ready for re-review.

**Pull request merged**